src/components/User/container.js

```diff
@@ -53,34 +53,36 @@ export default Container => class extends Component {
   sortingEvents = (events) => {
     const { createArray } = this;
     let eventArray = [];
-    const approvedArray = createArray(events, 'Approved');
-    const awaitApprovalArray = createArray(events, 'Awaiting approval');
-    const rejectedArray = createArray(events, 'Rejected');
-    const cancelledArray = createArray(events, 'Cancelled');
+    const approvedArray = createArray(events, eventDescription.APPROVED, eventTypes.ANNUAL_LEAVE);
+    const awaitApprovalArray = createArray(events, eventDescription.PENDING, eventTypes.ANNUAL_LEAVE);
+    const rejectedArray = createArray(events, eventDescription.REJECTED, eventTypes.ANNUAL_LEAVE);
+    const wfhArray = createArray(events, eventDescription.PENDING, eventTypes.WFH);
+    const cancelledArray = createArray(events, eventDescription.CANCELLED, eventTypes.ANNUAL_LEAVE);
 
     eventArray = [
-      { title: 'Approved', data: approvedArray },
-      { title: 'Awaiting approval', data: awaitApprovalArray },
-      { title: 'Rejected', data: rejectedArray },
-      { title: 'Cancelled', data: cancelledArray },
+      { title: eventDescription.APPROVED, data: approvedArray },
+      { title: eventDescription.PENDING, data: awaitApprovalArray },
+      { title: eventDescription.REJECTED, data: rejectedArray },
+      { title: eventTypes.WFH, data: wfhArray },
+      { title: eventDescription.CANCELLED, data: cancelledArray },
     ];
 
     return eventArray;
   }
 
-  createArray = (events, description) => {
+  createArray = (events, description, eventType) => {
     let arr = [];
-
-    arr.push(events.filter(event => event.eventStatus.description === description));
+    arr.push(events.filter(event => event.eventType.description === eventType
```

Comment:
indentation looks a bit weird here?

Reply:
yea I see what you mean. Does this look any better:

    arr.push(events
      .filter(event => event.eventType.description === eventType
      && event.eventStatus.description === description));